src/main/java/org/commonjava/util/sidecar/jaxrs/FoloContentAccessResource.java

```diff
@@ -87,11 +88,39 @@ public Uni<Response> get( @Parameter( in = PATH, required = true ) @PathParam( "
         Optional<File> download = archiveService.getLocally( path );
         if ( download.isPresent() && download.get().isFile() )
         {
-            InputStream inputStream = FileUtils.openInputStream( download.get() );
-            final Response.ResponseBuilder builder = Response.ok( new TransferStreamingOutput( inputStream ) );
-            logger.debug( "Download path: {} from historical archive.", path );
-            publishTrackingEvent( path, id );
-            return Uni.createFrom().item( builder.build() );
+            Uni<Boolean> checksumValidation =
+                    proxyService.validateChecksum( id, packageType, type, name, path, request );
```

Comment:
Does this mean there will be an individual request for each checksum ? can we make the folo records into the archive, and verify the checksum against that, to avoid the request to indy. Just an idea, not sure if that is possible.

Reply:
Hi, @sswguo
Local checksums will read the historical unpack payload directly, so cost would be small, and for remote checksums, will only download the necessary one(s) which local also owns, once validation is done, no other checksums will be downloaded.
For the folo record you suggest, I'm not sure if it will be retrieved before build is coming to end, since this happens at the progress of one build, and report should be still sealed in progress, the folo requests here should be direct to content requests with the corresponding content path, I'm afraid it will not have a better way than this since we need to know the new latest version of content. Corrent me if I'm wrong on this if we have a way to get the whole report on the current build.

Reply:
@sswguo This is a way to align the version between sidecar(local downloads) and Indy content retrieving, for the archive contents alignment with the current folo record of the finished build, it had some way done from Archive service.

Reply:
@yma96 thanks for the explanation, looks good for me. ; -)